emmarin/cl/cl/src/crust/note.rs

```diff
@@ -0,0 +1,135 @@
+use crate::crust::{balance::Unit, nullifier::NullifierCommitment};
+use crate::mantle::ZoneId;
+use crate::{Digest, Hash};
+use rand::RngCore;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub struct NoteCommitment(pub [u8; 32]);
+
+impl NoteCommitment {
+    pub fn as_bytes(&self) -> &[u8; 32] {
+        &self.0
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Serialize, Deserialize)]
+pub struct NoteWitness {
+    pub value: u64,
+    pub unit: Unit,
```

Comment:
Thoughts on making this a UnitWitness?

Reply:
it's not needed by outputs (and is actually one more hash), maybe move the unit to input/output?